t/run-complex-model-with-filters.t

```diff
@@ -59,6 +59,43 @@ for <hr-inclNOxExtended hr-inclNOxExtendedWithFilters> -> $model-version {
         is (+%output-hash<Storage><tan_into_application>).round(.001), $tan-into-application.round(.001),
                 "Correct tan_into_application result: { (+%output-hash<Storage><tan_into_application>).round(.001) }";
 
+        # check balance for each "Stufe" (livestock, storage, application)
+        # input = output + loss + remaining  
+        if ($model-version eq "hr-inclNOxExtendedWithFilters") { 
+            ### check livestock ntot balance
+            my $balance-livestock-ntot =
+                # get n into housing + yard + grazing 
+                %output-hash<Livestock><n_excretion>.apply-pairwise(
+                    # subtract nh3 loss from housing + yard + grazing
+                    %output-hash<Livestock><nh3_nlivestock>, &infix:<->, 0
+                ).apply-pairwise(
+                    # subtract n2 loss from grazing (housing + yard -> storage)
+                    %output-hash<Livestock><n2_ngrazing>, &infix:<->, 0
+                ).apply-pairwise(
+                    # subtract no loss from grazing (housing + yard -> storage)
+                    %output-hash<Livestock><no_ngrazing>, &infix:<->, 0
+                ).apply-pairwise(
+                    # subtract n2o loss from grazing (housing + yard -> storage)
+                    %output-hash<Livestock><n2o_ngrazing>, &infix:<->, 0
+                ).apply-pairwise(
+                    # subtract ntot remaining in soil from grazing
+                    %output-hash<Livestock><n_remain_grazing>, &infix:<->, 0
+                ).apply-pairwise(
+                    # subtract ntot remaining (vanishing) in air scrubber
+                    %output-hash<Livestock><tan_remain_scrubber>, &infix:<->, 0
+                ).apply-pairwise(
+                    # subtract ntot out of housing + yard + grazing
+                    %output-hash<Livestock><n_out_livestock>, &infix:<->, 0
+                );
```

Comment:
This could be written less repetitively with a loop:

```
my $balance-livestock-ntot = %output-hash<Livestock><n_excretion>;
for <nh3_nlivestock n2_ngrazing no_ngrazing ...> -> $input { # Add the others here
    $balance-livestock-ntot .= apply-pairwise(%output-hash<Livestock>{$input}, &infix:<->, 0);
}
```

Or if wanting to comment each line an array setup like:

```
my $balance-livestock-ntot = %output-hash<Livestock><n_excretion>;
my @inputs =
    'nh3_nlivestock', # nh3 loss from housing + yard + grazing
    'n2_ngrazing',    # n2 loss from grazing (housing + yard -> storage)
    ...;
for @inputs -> $input { # Add the others here
    $balance-livestock-ntot .= apply-pairwise(%output-hash<Livestock>{$input}, &infix:<->, 0);
}
```

@zaucker There's a shortcut syntax `&[-]` for infixes, but I think `&infix:<->` is probably a bit more clear in intent.